pkg/scheduler/core/util.go

```diff
@@ -91,11 +93,13 @@ func findOutScheduledCluster(tcs []workv1alpha2.TargetCluster, candidates []*clu
 		for _, cluster := range candidates {
 			if targetCluster.Name == cluster.Name {
 				res.Insert(targetCluster.Name)
+				validTarget = append(validTarget, targetCluster)
 				break
 			}
 		}
 	}
-	return res
+
+	return res, validTarget
```

Comment:
The `res` could be calculated by `validTarget`, I suggest remove it.
```diff
diff --git a/pkg/scheduler/core/division_algorithm.go b/pkg/scheduler/core/division_algorithm.go
index e0b2c082..689b8231 100644
--- a/pkg/scheduler/core/division_algorithm.go
+++ b/pkg/scheduler/core/division_algorithm.go
@@ -210,7 +210,7 @@ func scaleUpScheduleByReplicaDivisionPreference(
        preference policyv1alpha1.ReplicaDivisionPreference,
 ) ([]workv1alpha2.TargetCluster, error) {
        // Step 1: Find the clusters that have old replicas, so we can prefer to assign new replicas towards them.
-       scheduledClusterNames, scheduledClusters := findOutScheduledCluster(spec.Clusters, clusters)
+       scheduledClusters := findOutScheduledCluster(spec.Clusters, clusters)

        // Step 2: calculate the assigned Replicas in scheduledClusters
        assignedReplicas := util.GetSumOfReplicas(scheduledClusters)
@@ -229,7 +229,7 @@ func scaleUpScheduleByReplicaDivisionPreference(
        // If not, the old replicas may be recreated which is not expected during scaling up.
        // The parameter `scheduledClusterNames` is used to make sure that we assign new replicas to them preferentially
        // so that all the replicas are aggregated.
-       result, err := divideReplicasByPreference(clusterAvailableReplicas, newSpec.Replicas, preference, scheduledClusterNames)
+       result, err := divideReplicasByPreference(clusterAvailableReplicas, newSpec.Replicas, preference, util.ConvertToClusterNames(scheduledClusters))
        if err != nil {
                return result, err
        }
diff --git a/pkg/scheduler/core/util.go b/pkg/scheduler/core/util.go
index 1ceab827..93e2b8e4 100644
--- a/pkg/scheduler/core/util.go
+++ b/pkg/scheduler/core/util.go
@@ -77,11 +77,10 @@ func calAvailableReplicas(clusters []*clusterv1alpha1.Cluster, spec *workv1alpha

 // findOutScheduledCluster will return a name set of clusters
 // which are a part of `feasibleClusters` and have non-zero replicas.
-func findOutScheduledCluster(tcs []workv1alpha2.TargetCluster, candidates []*clusterv1alpha1.Cluster) (sets.String, []workv1alpha2.TargetCluster) {
+func findOutScheduledCluster(tcs []workv1alpha2.TargetCluster, candidates []*clusterv1alpha1.Cluster) []workv1alpha2.TargetCluster {
        validTarget := make([]workv1alpha2.TargetCluster, 0)
-       res := sets.NewString()
        if len(tcs) == 0 {
-               return res, validTarget
+               return validTarget
        }

        for _, targetCluster := range tcs {
@@ -92,14 +91,13 @@ func findOutScheduledCluster(tcs []workv1alpha2.TargetCluster, candidates []*clu
                // must in `candidates`
                for _, cluster := range candidates {
                        if targetCluster.Name == cluster.Name {
-                               res.Insert(targetCluster.Name)
                                validTarget = append(validTarget, targetCluster)
                                break
                        }
                }
        }

-       return res, validTarget
+       return validTarget
 }

 // resortClusterList is used to make sure scheduledClusterNames are in front of the other clusters in the list of
```

Reply:
it is better, and I have fixed it